lib/spack/spack/subprocess_context.py

```diff
@@ -94,17 +94,27 @@ def __init__(self):
             self.platform = spack.architecture.platform
             self.test_patches = store_patches()
 
-            # TODO: transfer spack.store.store? note that you should not
-            # transfer spack.store.store and spack.store.db: 'db' is a
-            # shortcut that accesses the store (so transferring both can
-            # create an inconsistency). Some tests set 'db' directly, and
-            # others set 'store'
+            self.store_root = spack.store.store.root
```

Comment:
(question) Out of curiosity, have you tried assigning `self.store = spack.store.store` (instead of saving the properties manually)? I wouldn't be surprised if that failed (i.e. if some object in the hierarchy was not pickle-able) and we already do "manual pickling" for other objects (e.g. `repo_dirs` in this function) so it's not an issue.

Reply:
Yes, I tried that. It failed and, to be honest, I didn't investigate it much since I saw that a `Store` object could be re-created from strings and I thought those were surely pickleable.